charts/elasticsearch/templates/backup/cronjob.yaml

```diff
@@ -0,0 +1,36 @@
+{{- if .Values.backup.enabled }}
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ .Values.name }}-backup-cronjob
+  labels: {{ include "elasticsearch.labels" . | nindent 4 }}
+spec:
+  schedule: "0 8 * * *" # Backup every day at 3 am EST
+  concurrencyPolicy: Replace
+  jobTemplate:
+    metadata:
+      labels: {{ include "elasticsearch.labels" . | nindent 8 }}
+    spec:
+      activeDeadlineSeconds: {{ .Values.backup.timeoutSeconds }}
+      template:
+        metadata:
+          labels: {{ include "elasticsearch.labels" . | nindent 12 }}
+        spec:
+          serviceAccountName: {{ .Values.name }}-backup-sa
+          restartPolicy: OnFailure
+          nodeSelector: {{ .Values.backup.nodeSelector | toYaml | nindent 12 }}
+          tolerations: {{ .Values.backup.tolerations | toYaml | nindent 12 }}
+          containers:
+          - name: {{ .Values.name }}-backup
+            image: {{ .Values.backup.imageRepo}}:{{ .Values.backup.imageTag }}
+            command: ['sh', '-c']
+            args:
+            - |-
+              TIMESTAMP=$( date "+%Y%m%d.%H%M%S" ) && \
```

Comment:
Unsure how this might be different for what looks like Bash inside a yaml file entry, but date format strings are usually single-quoted, and the `+` sign should appear before the opening quote (unless you intend it to be printed as a literal).

Example for your case: `+'%Y%m%d.%H%M%S'`

Reply:
TIL